examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,602 @@
+#
+# We can check the coefficient variability through cross-validation.
+# If coefficients vary in a significant way changing the input dataset
+# their robustness is not guaranteed, and they should probably be interpreted
+# with caution.
+
+from sklearn.model_selection import cross_validate
+from sklearn.model_selection import RepeatedKFold
+
+cv_model = cross_validate(
+    model, X, y, cv=RepeatedKFold(n_splits=5, n_repeats=5),
+    return_estimator=True, n_jobs=-1
+)
+coefs = pd.DataFrame(
+    [est.named_steps['transformedtargetregressor'].regressor_.coef_ *
+     X_train_preprocessed.std(axis=0)
+     for est in cv_model['estimator']],
+    columns=feature_names
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='cyan', saturation=0.5)
+plt.axvline(x=0, color='.5')
+plt.xlabel('Coefficient importance')
+plt.title('Coefficient importance and its variability')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The problem of correlated variables
+# -----------------------------------
+#
+# The AGE and EXPERIENCE coefficients are affected by strong variability which
+# might be due to the collinearity between the 2 features: as AGE and
+# EXPERIENCE vary together in the data, their effect is difficult to tease
+# apart.
+# To verify this interpretation we plot the variability of the AGE and
+# EXPERIENCE coefficient.
+
+plt.ylabel('Age coefficient')
+plt.xlabel('Experience coefficient')
+plt.grid(True)
+plt.scatter(coefs["AGE"], coefs["EXPERIENCE"])
+plt.title('Co-variations of coefficients for AGE and EXPERIENCE across folds')
+
+###############################################################################
+# Two regions are populated: when the EXPERIENCE coefficient is
+# positive the AGE one is negative and viceversa.
+#
+# To go further we remove one of the 2 features and check what is the impact
+# on the model stability.
+
+column_to_drop = ['AGE']
+
+cv_model = cross_validate(
+    model, X.drop(columns=column_to_drop), y,
+    cv=RepeatedKFold(n_splits=5, n_repeats=5),
+    return_estimator=True, n_jobs=-1
+)
+coefs = pd.DataFrame(
+    [est.named_steps['transformedtargetregressor'].regressor_.coef_ *
+     X_train_preprocessed.drop(columns=column_to_drop).std(axis=0)
+     for est in cv_model['estimator']],
+    columns=feature_names[:-1]
+)
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefs, orient='h', color='k', alpha=0.5)
+sns.boxplot(data=coefs, orient='h', color='cyan', saturation=0.5)
+plt.axvline(x=0, color='.5')
+plt.title('Coefficient importance and its variability')
+plt.xlabel('Coefficient importance')
+plt.subplots_adjust(left=.3)
+
+###############################################################################
+# The estimation of the EXPERIENCE coefficient is now less variable and
+# remain important for all predictors trained during cross-validation.
```

Comment:
Let's avoid the ambiguous "predictors". Maybe "models" or"estimators"

Reply:
I pick "models"